Fixes #2465 - Simplify/Enforce LayoutStyle.Absolute when X, Y, Width, and/or Height are null or Absolute()

tig opened the pull request:
Fixes #2465

BDisp commented on Terminal.Gui/Core/View.cs:
```diff
@@ -446,70 +446,102 @@ void SetTabIndex ()
 			get => frame;
 			set {
 				frame = new Rect (value.X, value.Y, Math.Max (value.Width, 0), Math.Max (value.Height, 0));
+				X = frame.X;
+				Y = frame.Y;
+				Width = frame.Width;
+				Height = frame.Height;
```
You are forcing the Pos/Dim to be `Absolute` even they aren't. So, in my opinion the `Frame` setter mustn't set the Pos/Dim values.

BDisp commented on Terminal.Gui/Core/View.cs:
```diff
@@ -446,70 +446,102 @@ void SetTabIndex ()
+					break;
+					
+				case LayoutStyle.Computed:
+					if (X == null) X = Frame.X;
+					if (Y == null) Y = Frame.Y;
+					if (Width == null) Width = Frame.Width;
+					if (Height == null) Height = Frame.Height;
```
Can be replaced by:
```cs
X ??= Frame.X;
Y ??= Frame.Y;
Width ??= Frame.Width;
Height ??= Frame.Height;
```
Additionally in each` Pos/Dim (X, Y, Width, Height)` setter you after must update the `_layoutStyle` field by get the `LayoutStyle` value:
`
_layoutStyle = LayoutStyle;
`

tig commented:
@BDisp and @tznind - I've finally come back to this PR and:

- [x] Merged with v2_develop
- [x] Updated API docs to be more complete and accurate
- [x] Fixed all Views that were using `Bounds` or calling `LayoutSubViews` before `IsInitalized`
- [x] Fixed a few other things (esp in ComboBox) that needed fixing

What I *didn't* do:

- [ ] Actually enforce LayoutStyle as was originally proposed here. I'd love your thoughts on that now.

Please review asap. Thanks!

BDisp commented:
Something is wrong with `All Views Tester`. It starting at (0,0) but navigate down and then up the views don't start at (0,0).

tig commented:
> Something is wrong with `All Views Tester`. It starting at (0,0) but navigate down and then up the views don't start at (0,0).

Ah, with mouse. With keyboard it's fine. On it...

BDisp commented:
> > Something is wrong with `All Views Tester`. It starting at (0,0) but navigate down and then up the views don't start at (0,0).
> 
> Ah, with mouse. With keyboard it's fine. On it...

With the keyboard. Navigate downward until `Dialog` and then navigate upward and you will see that the views aren't at (0,0) anymore.

tig commented:
> > > Something is wrong with `All Views Tester`. It starting at (0,0) but navigate down and then up the views don't start at (0,0).
> > 
> > 
> > Ah, with mouse. With keyboard it's fine. On it...
> 
> With the keyboard. Navigate downward until `Dialog` and then navigate upward and you will see that the views aren't at (0,0) anymore.
> 
> **Edit:** `Dialog` is centered horizontal and vertical and thus it seems the upper left coordinate is changed for all views.

Fixed.

BDisp commented:
Some of the views aren't show, like the `TextField`.

tig commented:
> Some of the views aren't show, like the `TextField`.

Fixed. ;-)

tig commented:
@BDisp I'd like to merge this so it can be picked up in https://github.com/gui-cs/Terminal.Gui/pull/3078 and https://github.com/BDisp/Terminal.Gui/pull/172 so I can more easily debug (this PR removes a ton of debug spew).

Are you good with my changes here?

BDisp commented:
Done @tig. Thanks.
